### PR TITLE
Upgrade libmbedtls to version 2.27.0

### PR DESCRIPTION
**Pull request by xyzhao2018**
Signed-off-by: xzhao3 <shirley.zhao@intel.com>

### CODE REVIEW

**Comment by GangSecurity** [comment]
you can port all the patches form 2.22.0 to 2.27.0, not just copy the code and make a patch.

**Review by jingdlu — COMMENTED** [review]
How do you get all of these changes?

**Review by jingdlu — COMMENTED** [review]
Two questions:
1, Is this original patch "Squashed commit upgrading to mbedtls-2.27.0" ?
2, I saw lots of changes regarding libtomcrypt. Do we still them?

**Review by GangSecurity — APPROVED** [review]

File: core/lib/libtomcrypt/mpi_desc.c [hunk 1]
```diff
@@ -545,8 +545,7 @@ static int montgomery_reduce(void *a, void *b, void *c)
 	if (mbedtls_mpi_grow(&A, N->n + 1))
 		goto out;
 
-	if (mbedtls_mpi_montred(&A, N, *mm, &T))
-		goto out;
+	mbedtls_mpi_montred(&A, N, *mm, &T);
```
Comment (jingdlu) [inline core/lib/libtomcrypt/mpi_desc.c:548]: This change should be in the commit "Squashed commit upgrading to mbedtls-2.27.0"? Right?

File: lib/libmbedtls/mbedtls/library/ssl_tls.c [hunk 2]
```diff
@@ -187,6 +187,10 @@ int mbedtls_ssl_session_copy( mbedtls_ssl_session *dst,
     mbedtls_ssl_session_free( dst );
     memcpy( dst, src, sizeof( mbedtls_ssl_session ) );
 
+#if defined(MBEDTLS_SSL_SESSION_TICKETS) && defined(MBEDTLS_SSL_CLI_C)
```
Comment (jingdlu) [inline lib/libmbedtls/mbedtls/library/ssl_tls.c:190]: all of this change come from CVE patch? If yes, please remove your signed-off-by.

File: core/lib/libtomcrypt/mpi_desc.c (hunk 1 above)
Comment (xyzhao2018) [reply core/lib/libtomcrypt/mpi_desc.c:548]: NO，Squashed commit upgrading to mbedtls-2.27.0 this patch is from CVE. The update above is not included in the patch.

File: lib/libmbedtls/mbedtls/library/ssl_tls.c (hunk 2 above)
Comment (xyzhao2018) [reply lib/libmbedtls/mbedtls/library/ssl_tls.c:190]: No, this CVE patch can't be used directly, we patch it manually. So two names are remained.

File: core/lib/libtomcrypt/mpi_desc.c (hunk 1 above)
Comment (jingdlu) [reply core/lib/libtomcrypt/mpi_desc.c:548]: Please check this commit: https://github.com/OP-TEE/optee_os/commit/7901324d9530594155991c8b283023d567741cc7

File: lib/libmbedtls/mbedtls/library/ssl_tls.c (hunk 2 above)
Comment (jingdlu) [reply lib/libmbedtls/mbedtls/library/ssl_tls.c:190]: Then can we just change the code manually? I mean just append your signed-off-by and try to make the title shorter.

Comment (xyzhao2018) [reply lib/libmbedtls/mbedtls/library/ssl_tls.c:190]: @GangSecurity , how to handle this case previously? Is there any precedents？

**Review by jingdlu — APPROVED** [review]

**Pull request merged**